### DISCUSSION TITLE
Created Timestamp Injection Failure Modes from Design Document

### DISCUSSION THREAD

**Discussion by swar8080** [prometheus/prometheus#14001] [Q&A]
Hello, the Created Timestamp [design document](https://docs.google.com/document/d/1kakDVn8aP1JerimLeazuMfy2jaB14W2kZDHMRokPcv4/edit#bookmark=id.mi9l494zy98y) mentions cases where CT is not appended. I'm wondering if these cases are still relevant or any new cases where added since then?

Specifically for `CT < H.startTimestamp`, is it possible that zero-injection is skipped if the CT is slightly before the creation of a new head block? Or did you end up implementing some tolerance around how old the CT is compared to the block start time?
<img width="869" alt="image" src="https://github.com/prometheus/prometheus/assets/17691679/8c7ad441-fc05-4256-a8ed-a8ebc9fd8c31">

**Comment by ArthurSens** [comment]
Hey 👋 ,

At the moment the limitations shown in the design document are still present. 

In this same design document, we talk about another approach that focuses on transforming Created Timestamps into metric metadata. With the metadata strategy, we would solve most of the limitations and we plan to start working on it next week :)

@bwplotka, @TheSpiritXIII and I will mentor a student through Google Summer of Code to solve most pending issues around Created Timestamps
